plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3015,40 +3015,42 @@ public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTabl
                 .map(systemTable -> new ClassLoaderSafeSystemTable(systemTable, getClass().getClassLoader()));
     }
 
-    private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName tableName)
+    private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName systemTableName)
     {
-        if (DeltaLakeTableName.isDataTable(tableName.getTableName())) {
+        Optional<DeltaLakeTableType> tableType = DeltaLakeTableName.tableTypeFrom(systemTableName.getTableName());
+        if (tableType.isEmpty() || tableType.get() == DeltaLakeTableType.DATA) {
             return Optional.empty();
         }
 
-        // Only when dealing with an actual system table proceed to retrieve the table handle
-        String name = DeltaLakeTableName.tableNameFrom(tableName.getTableName());
-        ConnectorTableHandle tableHandle;
+        String tableName = DeltaLakeTableName.tableNameFrom(systemTableName.getTableName());
+        Optional<DeltaMetastoreTable> table;
         try {
-            tableHandle = getTableHandle(session, new SchemaTableName(tableName.getSchemaName(), name));
+            table = metastore.getTable(systemTableName.getSchemaName(), tableName);
         }
         catch (NotADeltaLakeTableException e) {
-            // avoid dealing with non Delta Lake tables
             return Optional.empty();
         }
-        if (tableHandle == null) {
-            return Optional.empty();
-        }
-        if (tableHandle instanceof CorruptedDeltaLakeTableHandle) {
+        if (table.isEmpty()) {
             return Optional.empty();
         }
-        DeltaLakeTableHandle handle = (DeltaLakeTableHandle) tableHandle;
 
-        Optional<DeltaLakeTableType> tableType = DeltaLakeTableName.tableTypeFrom(tableName.getTableName());
-        if (tableType.isEmpty()) {
-            return Optional.empty();
+        String tableLocation = table.get().location();
+        TableSnapshot tableSnapshot = getSnapshot(new SchemaTableName(systemTableName.getSchemaName(), tableName), tableLocation, session);
+        try {
+            transactionLogAccess.getMetadataEntry(tableSnapshot, session);
         }
-        SchemaTableName systemTableName = new SchemaTableName(tableName.getSchemaName(), DeltaLakeTableName.tableNameWithType(name, tableType.get()));
+        catch (TrinoException e) {
+            if (e.getErrorCode().equals(DELTA_LAKE_INVALID_SCHEMA.toErrorCode())) {
+                return Optional.empty();
+            }
+            throw e;
+        }
```

Comment:
I know the purpose of this code because I have context to the PR.
For maintenance purposes I'd appreciate if these checks would be astracted

```suggestion
    if (isTableCorrupted(new SchemaTableName(systemTableName.getSchemaName(), tableName), tableLocation, session)) {
        return Optional.empty();
    }

Reply:
I'm not sure where `isTableCorrupted` method comes from. Can you elaborate on that?

Reply:
Sorry. I didn't express myself clearly.
I intended to suggest extracting the code to a method to encapsulate the logic related to seeing whether the table is or not corrupt.